c++/test/TestType.cc

```diff
@@ -490,4 +490,28 @@ namespace orc {
     *(footer.add_types()) = intType;
     checkProtoTypes(footer);
   }
+
+  TEST(TestType, testExistSubType) {
```

Comment:
Could you also add a roundtrip test to write and read ORC file without timestamp type with an invalid TZDIR env set? Something like we did in `TEST(TestTimezone, testMissingTZDB)` and be careful about the timezone cache.